DUI3-DX/Converters/Revit/Speckle.Converters.RevitShared/RevitConverterToSpeckle.cs

```diff
@@ -1,26 +1,62 @@
-using Autodesk.Revit.DB;
-using Speckle.Autofac.DependencyInjection;
+using Autofac.Features.Indexed;
+using System;
 using Speckle.Converters.Common;
 using Speckle.Converters.Common.Objects;
 using Speckle.Core.Models;
+using Autodesk.Revit.DB;
+using Speckle.Converters.RevitShared.Helpers;
 
 namespace Speckle.Converters.RevitShared;
 
 // POC: maybe possible to restrict the access so this cannot be created directly?
 public class RevitConverterToSpeckle : ISpeckleConverterToSpeckle
 {
-  private readonly IFactory<string, IHostObjectToSpeckleConversion> _toSpeckle;
+  private readonly IIndex<Type, IHostObjectToSpeckleConversion> _hostObjectConversions;
+  private readonly ToSpeckleConvertedObjectsCache _convertedObjectsCache;
+  private readonly ParameterValueExtractor _parameterValueExtractor;
 
-  public RevitConverterToSpeckle(IFactory<string, IHostObjectToSpeckleConversion> toSpeckle)
+  public RevitConverterToSpeckle(
+    IIndex<Type, IHostObjectToSpeckleConversion> hostObjectConversions,
+    ToSpeckleConvertedObjectsCache convertedObjectsCache,
+    ParameterValueExtractor parameterValueExtractor
+  )
   {
-    _toSpeckle = toSpeckle;
+    _hostObjectConversions = hostObjectConversions;
+    _convertedObjectsCache = convertedObjectsCache;
+    _parameterValueExtractor = parameterValueExtractor;
   }
 
   public Base Convert(object target)
   {
-    var objectConverter = _toSpeckle.ResolveInstance(nameof(Floor));
+    var objectConverter =
+      RetrieveObjectConversion(target.GetType())
+      ?? throw new SpeckleConversionException($"Could not find conversion for object of type {target.GetType()}");
+
+    Base result =
+      objectConverter.Convert(target)
+      ?? throw new SpeckleConversionException($"Conversion of object with type {target.GetType()} returned null");
+
+    // POC : where should logic common to most objects go?
+    if (target is Element element)
+    {
+      _convertedObjectsCache.AddConvertedBase(element.UniqueId, result);
+      _parameterValueExtractor.RemoveUniqueId(element.UniqueId);
+    }
+
+    return result;
+  }
+
+  private IHostObjectToSpeckleConversion? RetrieveObjectConversion(Type objectType)
```

Comment:
Interesting, we're looking for the specific type and then the subclass hmmm. I wonder if there is more here to exploit/scaffold

Reply:
yeah, I'm wondering if we can make conversion "pipelines" or something like that where we could add all the Wall properties like length and height, and then continue passing the object down to more generic conversions that could add more generic info like parameters or display value.

Reply:
We should look into this idea Connor. I think it's achievable and makes more sense than having to add parameters on every conversion